milli/src/search/new/bucket_sort.rs

```diff
@@ -154,6 +160,28 @@ pub fn bucket_sort<'ctx, Q: RankingRuleQueryTrait>(
     }
 
     while valid_docids.len() < length {
+        if time_budget.exceeded() {
+            loop {
+                let bucket = std::mem::take(&mut ranking_rule_universes[cur_ranking_rule_index]);
+                ranking_rule_scores.push(ScoreDetails::Skipped);
+                maybe_add_to_results!(bucket);
+                ranking_rule_scores.pop();
+
+                if cur_ranking_rule_index == 0 {
```

Comment:
This is already handled inside of `back!`, although allegedly this is a bit dangerous as the `break` in `back!` is not intended to break the same loop.

Reply:
Yes, I was not super comfortable relying on this mechanism (or even on the `back!` macro at all tbh)

Reply:
Agreed